app/models/firmware_registry/rest_api_depot.rb

```diff
@@ -31,6 +31,27 @@ def sync_fw_binaries_raw
     end
   end
 
+  def self.do_create_firmware_registry(options)
+    transaction do
```

Comment:
The transaction feels like overkill here but don't really see anything wrong with it